source/agora/node/FullNode.d

```diff
@@ -1186,6 +1186,8 @@ public class FullNode : API
 
     private void pushBlock (const Block block) @trusted
     {
+        if (this.is_shutting_down)
+            return;
```

Comment:
This cannot prevent accessing the array after it being `null` because the flow could already be in the `foreach`, for example. So it would be better to not set `null` to the arrays in the `shutdown` as not setting `null` to the arrays makes any harm, I think.

Reply:
Thanks I created an issue to follow up on this.